subsys/pm/policy/policy_state_lock.c

```diff
@@ -121,7 +135,8 @@ bool pm_policy_state_is_available(enum pm_state state, uint8_t substate_id)
 		if (substates[i].state == state &&
 		   (substates[i].substate_id == substate_id || substate_id == PM_ALL_SUBSTATES)) {
 			return (atomic_get(&lock_cnt[i]) == 0) &&
-			       (atomic_get(&latency_mask) & BIT(i));
```

Comment:
I'm thinking that in the case of a global lock for all states and substates, it would be better from a performance perspective to check the global lock first and return early.

```c
bool pm_policy_state_is_available(enum pm_state state, uint8_t substate_id)
{
#if DT_HAS_COMPAT_STATUS_OKAY(zephyr_power_state)
    /* Early return if all states are globally locked - most efficient */
    if (atomic_get(&global_lock_cnt) != 0) {
        return false;
    }
    
   
    for (size_t i = 0; i < ARRAY_SIZE(substates); i++) {
     /* Continue with existing logic only if global lock is not active */
    }
#endif
    return false;
}
```